browser-test/src/support/session_timeout.ts

```diff
@@ -0,0 +1,228 @@
+
+      // Return adjusted data
+      return {
+        inactivityWarning: data.inactivityWarning + clockSkew,
+        inactivityTimeout: data.inactivityTimeout + clockSkew,
+        totalWarning: data.totalWarning + clockSkew,
+        totalTimeout: data.totalTimeout + clockSkew,
+        currentTime: data.currentTime,
+      }
+    } catch (e) {
+      console.error('Failed to parse timeout cookie:', e)
+      return null
+    }
+  })
+}
+
+/**
+ * Helper function to mock the Date object and advance time
+ */
+async function mockDateAndAdvanceTime(page: Page, advanceMilliseconds: number) {
+  await page.evaluate((ms) => {
+    // Store the original Date constructor
+    const OriginalDate = Date
+    const originalNow = Date.now
+
+    // Create a mock date that will be ahead by the specified milliseconds
+    const mockTimeOffset = ms
+
+    const realNow = originalNow.call(Date)
+    const mockedNow = realNow + mockTimeOffset
+
+    console.log(
+      `Mocking time: Real time: ${realNow / 1000}, Mocked time: ${mockedNow / 1000}, Offset: ${mockTimeOffset / 1000}`,
+    )
+
+    // Override the Date constructor
+    // @ts-expect-error - Intentionally mocking Date constructor
+    window.Date = class extends OriginalDate {
+      constructor(...args: ConstructorParameters<typeof OriginalDate>) {
+        if (args.length > 0) {
+          // If arguments provided, behave normally
+          super(...args)
+        } else {
+          // When called with no arguments, return current time + offset
+          super(realNow + mockTimeOffset)
+        }
+      }
+
+      // Override getTime to return the advanced time
+      getTime() {
+        return realNow + mockTimeOffset
+      }
+
+      // Override now to return the advanced time
+      static now() {
+        return realNow + mockTimeOffset
+      }
+    }
+
+    // Dispatch timechange event with time details. Simply advancing time does not trigger the timeout
+    // that is already set up in the browser
+    const timeChangeEvent = new CustomEvent('timechange', {
+      detail: {
+        realTime: realNow,
+        mockedTime: mockedNow,
+        offset: mockTimeOffset,
+      },
+    })
+
+    console.log(
+      'Dispatching timechange event with details:',
+      timeChangeEvent.detail,
+    )
+    window.dispatchEvent(timeChangeEvent)
+  }, advanceMilliseconds)
+  // Give the browser a moment to process the event
+  await page.waitForTimeout(100)
+}
+
+/**
+ * Advances time to trigger a specific timeout event
+ */
+async function advanceToTimeout(page: Page, timeoutType: TimeoutType) {
+  // Get the adjusted timeout data that accounts for clock skew
+  const data = await getAdjustedTimeoutData(page)
+  if (!data) {
+    throw new Error('No timeout data found')
+  }
+
+  // Get current browser time
+  const browserNow = await page.evaluate(() => Math.floor(Date.now() / 1000))
+
+  // Get target time based on timeout type
+  const targetTime = data[timeoutType]
```

Comment:
```suggestion
  const targetTime = data[timeoutType]
```
You should be able to slot in the clock api here